src/extensionVariables.ts

```diff
@@ -21,4 +22,10 @@ export namespace ext {
     export let tagFS: TagFileSystem;
     export let diagnosticWatcher: Disposable | undefined;
     export let diagnosticCollection: DiagnosticCollection;
+
+    export const resourceProviders: { [key: string]: ApplicationResourceProvider } = {};
```

Comment:
Well, I thought all extensions would be considered resource providers including our Azure extensions.  Does the SWA extension not register its own resource to provide the SWA tree items?

Reply:
We need to be able to fetch Azure resources separately from the resource extensions (SWA). We need this capability so we can A. list resources quickly without waiting for the other extensions to activate, and B. show resources even if a user doesn't have the associated extension installed. 

So RGs will register an `AzureResourceProvider` which uses the arm-resources SDK to fetch resources.

See my implementation here:
https://github.com/microsoft/vscode-azureresourcegroups/blob/98ee1b6a61097c0ee97cf2ac539d99a4f33c8d0e/src/AzureResourceProvider.ts

Reply:
What SWA will register is a resolver that takes the generic resources fetched by RGs and the arm-resources SDK and "returns" StaticWebAppTreeItems.

See here: https://github.com/microsoft/vscode-azurestaticwebapps/blob/17dfbd82dd8acb167cfafc0472c22f3611109736/src/StaticWebAppResolver.ts

Reply:
Yeah, Alex is right. For now I'm expecting there to be only one resource provider (registered into itself by RG extension), with the feature extensions providing resolvers.

That being said, since _someday_ we might make it a map, why not just make it a map now?